src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -162,11 +162,11 @@ public Task StartAsync()
 
                     await _startTcs.Task;
                     await _receiveLoopTask;
-
                     _logger.LogDebug("Draining event queue");
-                    await _eventQueue.Drain();
 
-                    _httpClient.Dispose();
+                    await Task.WhenAny(
```

Comment:
@moozzyk preferred this, but he said that you would probably hate it because of the ContinueWith 😄